### PR TITLE
feat: allow re-assessments for examiners in a cyle id

### PR DESCRIPTION
**Pull request by tushar5526**
Migrations PR : https://github.com/Mission-Prerna/nl-hasura-migrations/pull/3

### CODE REVIEW

File: src/app.service.ts [hunk 1]
```diff
@@ -322,13 +322,61 @@ export class AppService {
   async createAssessmentVisitResult(
     createAssessmentVisitResultData: CreateAssessmentVisitResult,
   ) {
+
+    // First check if for examiner the student submitted is in cycle
```
Comment (karntrehan) [inline src/app.service.ts:326]: Extract function

File: src/admin/admin.service.ts [hunk 2]
```diff
@@ -605,37 +605,85 @@ export class AdminService {
...
           udise: {
             in: data.udises,
           },
-          mentor_id: data.mentor_id,
-          cycle_id: cycleId,
         },
-      }));
+      })]
 
+      if(data.reset_all) {
```
Comment (karntrehan) [inline src/admin/admin.service.ts:677]: Extract function

Comment (tushar5526) [reply src/admin/admin.service.ts:677]: This is already extracted in a sub-function which starts at 666 and ends at 683. 

684 is where we add the subfunction to list of promises.

File: src/app.service.ts (hunk 1 above)
Comment (tushar5526) [reply src/app.service.ts]: done

**Pull request merged**